src/figuresClasses.ts

```diff
@@ -1,11 +1,82 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const biggestSide = Math.max(a, b, c);
+    const arr = [a, b, c];
+    let count = 0;
+
+    for (const item of arr) {
+      if (item !== biggestSide) {
+        count += item;
+      }
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0 || biggestSide >= count) {
+      throw new Error('your error message');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. Instead of 'your error message', provide a message that explains what went wrong, such as 'Triangle sides must be positive numbers and adhere to the triangle inequality theorem.'